compiler/src/dotty/tools/dotc/plugins/Plugin.scala

```diff
@@ -0,0 +1,176 @@
+package dotty.tools.dotc
+package plugins
+
+import core._
+import Contexts._
+import Phases._
+import dotty.tools.io._
+import transform.MegaPhase.MiniPhase
+
+import java.io.InputStream
+
+import scala.collection.mutable
+import scala.util.{ Try, Success, Failure }
+
+trait PluginPhase extends MiniPhase {
+  def runsBefore: Set[Class[_ <: Phase]] = Set.empty
```

Comment:
For inter-plugin dependencies, it will be impossible for a plugin A to get hold of the `Class[_]` value for a plugin B coming from a different jar (hence in a different `ClassLoader`, so not even reflectively available).

We need to use *names* here (also for `runsAfter`). Names are usually not as good as token identities, but here they offer an essential part of the expressiveness.